lib/Parse/ParseExpr.cpp

```diff
@@ -542,8 +542,9 @@ ParserResult<Expr> Parser::parseExprSequenceElement(Diag<> message,
     consumeToken();
   }
 
-  // Try to parse '@' sign or 'inout' as a attributed typerepr.
-  if (Tok.isAny(tok::at_sign, tok::kw_inout)) {
+  // Try to parse '@' sign, 'inout' or 'nonisolated' as a attributed typerepr.
+  if (Tok.isAny(tok::at_sign, tok::kw_inout) ||
+      Tok.isContextualKeyword("nonisolated")) {
```

Comment:
Hmm, does this mean the following tries to parse as a `nonisolated` type?

```swift
func nonisolated(_ x: Int) {}
nonisolated(0)
print("hello")
```

I noticed the swift-syntax PR doesn't include similar logic, should this just be removed?

Reply:
If I remove this condition C++ parser cannot parse - `let _ = [nonisolated(nonsending) () async -> Void]()`, maybe there is a better way to do this I don't know about...

Reply:
I tightened the condition in `canParseNonisolatedAsTypeModifier` and added your snippet. Rintaro pointed me to this spot initially for `[nonisolated(nonsending) ...]()` support, but if there is a better way please let me know.